src/core/api.c

```diff
@@ -1321,11 +1321,13 @@ MsQuicStreamReceiveComplete(
         Stream,
         BufferLength);
 
+    CxPlatRwLockAcquireShared(&Stream->ReceiveCompleteInlineLock);
```

Comment:
I would like to try to do this without a lock. Let me think if there's a way we might be able to do that...